gatorgrade/input/in_file_path.py

```diff
@@ -7,3 +7,28 @@ def parse_yaml_file(file_path):
     with open(file_path, encoding="utf8") as file:
         data = yaml.load_all(file, Loader=yaml.FullLoader)
         return list(data)
+
+
+def reformat_yaml_data(data):
+    """Reformat the raw data from a YAML file into a list of tuples."""
+    reformatted_data = []
+    data.pop(0)  # Removes the setup commands
+    loop_through_data(None, data[0], reformatted_data)
+    return reformatted_data
+
+
+def loop_through_data(path, data_list, reformatted_data):
+    """Recursively loop through the data and add any checks that are found to the reformatted list."""
+    current_path = path
+    for dict in data_list:
+        for item in dict:
+            if str(type(dict[item])) == "<class 'list'>":
```

Comment:
there is a better way to check for this condition. You can use the `isinstance` function to write something like this:

```python
if isinstance(dict[item], list):

```